soh/src/code/z_parameter.c

```diff
@@ -2305,6 +2305,8 @@ u8 Item_Give(PlayState* play, u8 item) {
                         }
                     } else {
                         gSaveContext.equips.buttonItems[i] = ITEM_NONE;
+                        // No need to call OnItemGive for ITEM_SOLD_OUT. Doesn't have any ItemTable entries anyway.
+                        return -1;
```

Comment:
This feels wrong for a couple reasons:
* `Item_Give` is a `u8`, so returning `-1` is bad
* It seems like this should be able to be addressed in `Return_Item` instead

Reply:
See my conversation on this exact line with leggett above.

Reply:
Also, everything else has an ItemTable entry. However, I can see wanting to cover all future bases.